main.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+    "flag"
+    "fmt"
+    "./schemas"
+    "github.com/companieshouse/chs.go/avro"
+    "github.com/companieshouse/chs.go/kafka/producer"
+    "github.com/companieshouse/chs.go/avro/schema"
+    "os"
+    "os/signal"
+    "github.com/Shopify/sarama"
+    "strings"
+    "strconv"
+)
+
+// Assigns all flags to variables
+var (
+    brokerPtr = flag.String("broker", "", "Broker")
+    topicPtr = flag.String("topic", "", "Topic name")
+    schemaPtr = flag.String("schema", "", "Schema")
+    schemaRegistryPtr = flag.String("schema-registry", "", "Schema Registry")
+    partitionPtr = flag.Int64("partition", 0, "Partition")
+    zookeeperPtr = flag.String("zookeeper", "", "Zookeeper")
+    offsetPtr = flag.String("offset", "-1", "Offset number")
+    jsonOutPtr = flag.Int("json-out", -1, "Does deserialized JSON get printed to terminal?")
+)
+
+func main () {
+    flag.Parse()
+
+    fmt.Println("broker:", *brokerPtr)
+    fmt.Println("topic:", *topicPtr)
+    fmt.Println("schema:", *schemaPtr)
+    fmt.Println("schema-registry:", *schemaRegistryPtr)
+    fmt.Println("partition:", *partitionPtr)
+    fmt.Println("zookeeper-registry:", *zookeeperPtr)
+    fmt.Println("offset:", *offsetPtr)
+    fmt.Println("json-out:", *jsonOutPtr)
```

Comment:
It's worth considering the use of `flag.Visit()` or `flag.VisitAll()` here, depending on whether the output should include only the flags that have been set or all flags regardless of value, as in this example:

```
flag.VisitAll(func (f *flag.Flag) {
    fmt.Printf("%v: %v\n", f.Name, f.Value)
})
```

Reply:
Good point, however if the flags that have only been set are outputted, the user may not know why the tool isn't working as it could be using a defaulted value and because it isn't printed out they'll never know?